server/sessionstore.go

```diff
@@ -63,41 +72,41 @@ func (ss *SessionStore) NewSession(conn interface{}, sid string) (*Session, int)
 		s.send = make(chan interface{}, sendQueueLimit+32) // buffered
 		s.stop = make(chan interface{}, 1)                 // Buffered by 1 just to make it non-blocking
 		s.detach = make(chan string, 64)                   // buffered
-		if globals.cluster != nil {
+
+		if globals.cluster != nil && s.proto != CLUSTER {
+			// This is only useful when running as a cluster and only for non-proxied sessions.
 			s.remoteSubs = make(map[string]*RemoteSubscription)
 		}
 	}
 
 	s.lastTouched = time.Now()
-	if s.sid == "" {
-		s.sid = store.GetUidString()
-	}
 
 	ss.lock.Lock()
 
-	ss.sessCache[s.sid] = &s
-	count := len(ss.sessCache)
-	var expired []*Session
 	if s.proto == LPOLL {
 		// Only LP sessions need to be sorted by last active
 		s.lpTracker = ss.lru.PushFront(&s)
+	}
 
-		// Remove expired sessions
-		expire := s.lastTouched.Add(-ss.lifeTime)
-		for elem := ss.lru.Back(); elem != nil; elem = ss.lru.Back() {
-			sess := elem.Value.(*Session)
-			if sess.lastTouched.Before(expire) {
-				ss.lru.Remove(elem)
-				delete(ss.sessCache, sess.sid)
-				expired = append(expired, sess)
-			} else {
-				break // don't need to traverse further
-			}
+	ss.sessCache[s.sid] = &s
+
+	// Expire stale long polling sessions.
```

Comment:
This is confusing from just reading the code.
One needs to remember that `SessionStore.lru` is only used for `LPOLL`. I suggest either adding a comment explaining it or moving this code back to the s.proto == LPOLL block above.

Reply:
Expanded the comment. 

The `s.proto == LPOLL` was what caused the problem. It limited cleaning up of stale LP sessions only to when a new LP session was initialized.